app/controllers/criteria_controller.rb

```diff
@@ -0,0 +1,32 @@
+class CriteriaController < ApplicationController
+
+  def new
+    @assignment = Assignment.find(params[:assignment_id])
+    @criterion_type = params[:criterion_type]
+    @criterion = get_class(@criterion_type).new
```

Comment:
I could avoid doing the check and creating the constants if I use:
@criterion = (@criterion_type.camelize + 'Criterion').constantize.new